### DISCUSSION TITLE
How to make Pocketbase work locally with HTTPS on Windows 10?

### DISCUSSION THREAD

**Discussion by suelitonx** [pocketbase/pocketbase#4390] [Q&A] [answered]
Hello everyone,

I'm new to Pocketbase development and I'm having trouble configuring Pocketbase locally with HTTPS on Windows 10. I would like to ask for the community's help for guidance on how to perform this configuration.

Would it be possible for someone to guide me step by step on how to set up Pocketbase to work locally with HTTPS support on Windows 10? I'm familiar with basic development concepts, but I'm a beginner in more advanced server configuration issues.

I appreciate any help or suggestions you can provide!

Thank you!

**Comment by ganigeorgiev** [comment, marked as answer]
PocketBase doesn't have a builtin mechanism to generate ~or load~ (_see the edit_) a self-signed certificate if that is what you are trying to do (_edit: with the latest releases you can load any certificate or custom TLS configuration via the `e.Server` and `e.TLSManager` fields of the `OnServe` hook event_).

If you want to use HTTPS on localhost just for test purposes (ex. because an OAuth2 provider requires it), then I'd recommend using a tunneling service like [ngrok](https://ngrok.com/download) or [zrok](https://docs.zrok.io/docs/getting-started/#installing-the-zrok-command).

```sh
# start PocketBase as usual
./pocketbase serve

# in a new terminal start the tunel for the PocketBase port
ngrok http 8090

# use the generated https address...
```
